docs/design.html

```diff
@@ -322,7 +322,7 @@ <h3 class="anchor-heading"><a id="replication" class="anchor-link"></a><a href="
     Followers consume messages from the leader just as a normal Kafka consumer would and apply them to their own log. Having the followers pull from the leader has the nice property of allowing the follower to naturally
     batch together log entries they are applying to their log.
     <p>
-    As with most distributed systems automatically handling failures requires having a precise definition of what it means for a node to be "alive." In Kafka, a special node
+    As with most distributed systems, automatically handling failures requires a precise definition for when a node is considered "alive." In Kafka, a special node
```

Comment:
Agree with the removal of "having." I am not sure about the latter half of the sentence. We're trying to suggest that the definition of liveness itself varies system by system, so we need to be precise when we talk about it. The extra emphasis in the original version seemed to convey that better. The use of "when" sounds like we're suggesting some kind of temporal uncertainty, but maybe I'm overanalyzing it.